### DISCUSSION TITLE
Does SolidStart support SSG with a remote data source?

### DISCUSSION THREAD

**Discussion by schnz** [solidjs/solid-start#920] [Q&A]
Pretty much what the title states. I want to retrieve data during build time from a GraphQL source (Headless CMS). Is that scenario supported somehow? I fear not, but wanted to ask the community before giving up on it.

**Comment by JaielZeus** [comment]
[Does the SSG site not help you on this issue? ](https://docs.solidjs.com/references/concepts/ssr/ssg)

If you need data at build time do an async/await routine...

**Reply by schnz** [reply]
I did give it a try. No success though. No matter where I try to execute the code, its always executed at runtime by the client or not at all. Do I miss something obvious here? 

- As top level async/await -> Executed by the client (and "node-fetch" import can't be used as it wouldn't run in the browser)
- Within createServerAction$ -> Would require SSR, but i want SSG.
- Within entry-client.tsx -> Also seems to be executed by the client

By now I switched to Astro which I found to be superior for my use case. But For the sake of curiosity I would still like to get this working.
